extensions/functions_datetime.yaml

```
@@ -265,3 +342,111 @@ scalar_functions:
           - name: y
             value: interval_year
         return: boolean
+  -
+    name: "assume_timezone"
+    description: Convert local time to UTC using given local time's timezone.
```

Technically, `timestamp_tz` doesn't need to be represented using UTC. It could just be an offset to whatever epoch the consumer likes best, as long as the epoch is well-defined. Obviously a sane implementation would just use an offset to the unix epoch, which is defined in UTC, but for as far as Substrait is concerned, UTC is pretty meaningless here. Maybe something like

```suggestion
    description: Convert a local timestamp represented with the given timezone to a (timezone-agnostic) timestamp_tz.
```

It should also specify what happens when the timestamp does not exist for the given timezone due to DST shenanigans. I guess it should just error out in that case.

Also, what about the reverse operation?

---

As per [`type_classes.md`](https://github.com/substrait-io/substrait/blob/main/site/docs/types/type_classes.md):
```
| timestamp_tz    | A timezone-aware timestamp within [1000-01-01 00:00:00.000000 UTC..9999-12-31 23:59:59.999999 UTC], with microsecond precision. Similar to aware datetime in Python. | `int64` microseconds since 1970-01-01 00:00:00.000000 UTC
```
I think keeping timestamp_tz in UTC would be best since (to my knowledge) nobody is doing anything else and local time can be ambiguous and non-existent.

---

I meant this more like "`timestamp_tz` represents a time delta compared to some epoch," which doesn't require the concept of timezones to even exist to be defined. I absolutely agree that every sane implementation should be using UTC when reasoning about what this time delta means and defining the epoch, but I could define my own timezone that's exactly pi days offset from UTC and "use that" for `timestamp_tz`. Nothing should break because its definition doesn't depend on any timezone definition.